### ISSUE TITLE
Heuristics for efficient state preparation using arbitrary 2q target gate

### ISSUE THREAD

**Issue by tanujkhattar** [quantumlib/Cirq#4721]
This came up in the review of https://github.com/quantumlib/Cirq/pull/4707 that we could potentially write a heuristic for efficient state preparation using a single application of an entangling 2q gate by solving for surrounding `PhsXZ` angles and optimizing the fidelity b/w current state and desired state. The approach would be similar to approximate unitary decomposition approach proposed in https://github.com/quantumlib/Cirq/issues/4362

P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by verult** [comment]
Marking as after 1.0 as this is a feature addition

**Comment by mhucka** [comment]
I'm closing this as a nice-to-have-but-we-don't-have-time feature request. If this is incorrect or something changes, someone please feel free to reopen it.
